src/Serializer.php

```diff
@@ -130,6 +142,34 @@ public function encodeMessage($message)
         }
     }
 
+    /**
+     * Encode rest metadata errors to the correct protobuf Error type
+     *
+     * @param array $metadata
+     * @return array
+     */
+    public static function encodeMetadataToProtobufErrors(array $metadata): array
```

Comment:
rather than have a different method for REST metadata, would it be possible to make the existing `decodeMetadata` handle both REST and gRPC metadata?

Reply:
I thought of that but the fact that REST received a completely different structure made me go this way. Handling everything on the `decodeMetadata` method which is already a kind of loaded method, felt a bit convoluted IMO.

Maybe there is a way though, but haven't thought of it.